nerfstudio/models/nerfacto.py

```diff
@@ -327,15 +322,20 @@ def get_outputs(self, ray_bundle: RayBundle):
     def get_metrics_dict(self, outputs, batch):
         metrics_dict = {}
         image = batch["image"].to(self.device)
-        metrics_dict["psnr"] = self.psnr(outputs["rgb"], image)
+        metrics_dict["psnr"] = self.psnr(outputs["rgb"], image[:, :3])
         if self.training:
             metrics_dict["distortion"] = distortion_loss(outputs["weights_list"], outputs["ray_samples_list"])
         return metrics_dict
 
     def get_loss_dict(self, outputs, batch, metrics_dict=None):
         loss_dict = {}
         image = batch["image"].to(self.device)
-        loss_dict["rgb_loss"] = self.rgb_loss(image, outputs["rgb"])
+        if outputs.get("background_color", None) is not None and image.shape[-1] == 4:
+            background_color = outputs["background_color"]
+            image_ = image[:, :3] * image[:, 3:] + background_color.to(image.device) * (1 - image[:, 3:])
+        else:
+            image_ = image[:, :3]
+        loss_dict["rgb_loss"] = self.rgb_loss(image_, outputs["rgb"])
```

Comment:
Can you add the same logic to [NGPModel](https://github.com/nerfstudio-project/nerfstudio/blob/main/nerfstudio/models/instant_ngp.py)? I found that this model works (at least with the `instant-ngp-bounded` default settings) works particularly well with alpha transparent training (at least for the masked mustard dataset). It might also be beneficial for other models but maybe not worth adding for the others until it becomes useful for people.